### PR TITLE
[IOS-8239] - extend ViewableRouter

### PR DESCRIPTION
**Pull request by DmitryPashinskiy**
- presentation transition block has been added

### CODE REVIEW

File: ios/RIBs/Classes/ViewableRouter.swift [hunk 1]
```diff
@@ -26,6 +26,13 @@ public protocol ViewableRouting: Routing {
 
     /// The base view controllable associated with this `Router`.
     var viewControllable: ViewControllable { get }
+
+    /// this block will be used as prior in case if it's not nil
+    var presentationTransition: ((_ routing:ViewableRouting,_ completion:(() -> Void)) -> Void)? { get }
```
Comment (kradenkov) [inline ios/RIBs/Classes/ViewableRouter.swift]: Have you run the linter? I suppose there will be warnings about spaces.

Comment (ValeriyMaliuk) [reply ios/RIBs/Classes/ViewableRouter.swift]: @kradenkov why? looks okay

Comment (DmitryPashinskiy) [inline ios/RIBs/Classes/ViewableRouter.swift:31]:
```suggestion
var presentationTransition: (( _ routing: ViewableRouting, _ completion: (() -> Void)) -> Void)? { get }
```

Comment (DmitryPashinskiy) [inline ios/RIBs/Classes/ViewableRouter.swift]:
```suggestion
    var presentationTransition: (( _ routing: ViewableRouting, _ completion: (() -> Void)) -> Void)? { get }
```

Comment (kradenkov) [reply ios/RIBs/Classes/ViewableRouter.swift]: Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)
Comma Spacing Violation: There should be no space before and one after any comma. (comma)

Comment (DmitryPashinskiy) [reply ios/RIBs/Classes/ViewableRouter.swift]: fixed

Comment (ValeriyMaliuk) [reply ios/RIBs/Classes/ViewableRouter.swift]: aaaaa

Comment (ValeriyMaliuk) [reply ios/RIBs/Classes/ViewableRouter.swift]: you are totally right

File: ios/RIBs/Classes/ViewableRouter.swift [hunk 2]
```diff
@@ -26,6 +26,13 @@ public protocol ViewableRouting: Routing {
 
     /// The base view controllable associated with this `Router`.
     var viewControllable: ViewControllable { get }
+
+    /// this block will be used as prior in case if it's not nil
+    var presentationTransition: (( _ routing: ViewableRouting, _ completion: (() -> Void)) -> Void)? { get }
+}
+
+extension ViewableRouting {
+    var presentationTransition: ((_ routing:ViewableRouting,_ completion:(() -> Void)) -> Void)? { return nil }
```
Comment (kradenkov) [inline ios/RIBs/Classes/ViewableRouter.swift]: the same

File: ios/RIBs/Classes/ViewableRouter.swift [hunk 3]
```diff
@@ -41,6 +48,9 @@ open class ViewableRouter<InteractorType, ViewControllerType>: Router<Interactor
     /// The base `ViewControllable` associated with this `Router`.
     public let viewControllable: ViewControllable
 
+    /// this block will be used as prior in case if it's not nil
+    public var presentationTransition: ((_ routing:ViewableRouting,_ completion:(() -> Void)) -> Void)?
```
Comment (kradenkov) [inline ios/RIBs/Classes/ViewableRouter.swift]: the same

**Review by ValeriyMaliuk — DISMISSED** [review]

File: ios/RIBs/Classes/ViewableRouter.swift (hunk 2 above)
Comment (DmitryPashinskiy) [inline ios/RIBs/Classes/ViewableRouter.swift]:
```suggestion
public extension ViewableRouting {
    var presentationTransition: ((_ routing: ViewableRouting, _ completion: (() -> Void)) -> Void)? { return nil }
```

Comment (DmitryPashinskiy) [reply ios/RIBs/Classes/ViewableRouter.swift]: fixed

File: ios/RIBs/Classes/ViewableRouter.swift (hunk 3 above)
Comment (DmitryPashinskiy) [inline ios/RIBs/Classes/ViewableRouter.swift]:
```suggestion
    public var presentationTransition: (( _ routing: ViewableRouting, _ completion: (() -> Void)) -> Void)?
```

Comment (DmitryPashinskiy) [reply ios/RIBs/Classes/ViewableRouter.swift]: fixed

**Review by kradenkov — APPROVED** [review]

**Review by ValeriyMaliuk — APPROVED** [review]

**Pull request merged**